GLTFSDK.Samples/Deserialize/Source/main.cpp

```diff
@@ -0,0 +1,304 @@
+        }
+
+        // Pass the absolute path, without the filename, to the stream reader
+        auto streamReader = std::make_unique<StreamReader>(path.parent_path());
+
+        std::experimental::filesystem::path pathFile = path.filename();
+        std::experimental::filesystem::path pathFileExt = pathFile.extension();
+
+        std::string manifest;
+
+        auto MakePathExt = [](const std::string& ext)
+        {
+            return "." + ext;
+        };
+
+        std::unique_ptr<GLTFResourceReader> resourceReader;
+
+        // If the file has a '.gltf' extension then create a GLTFResourceReader
+        if (pathFileExt == MakePathExt(GLTF_EXTENSION))
+        {
+            auto gltfStream = streamReader->GetInputStream(pathFile.u8string()); // Pass a UTF-8 encoded filename to GetInputString
+            auto gltfResourceReader = std::make_unique<GLTFResourceReader>(std::move(streamReader));
+
+            std::stringstream manifestStream;
+
+            // Read the contents of the glTF file into a string using a std::stringstream
+            manifestStream << gltfStream->rdbuf();
+            manifest = manifestStream.str();
+
+            resourceReader = std::move(gltfResourceReader);
+        }
+
+        // If the file has a '.glb' extension then create a GLBResourceReader. This class derives
+        // from GLTFResourceReader and adds support for reading manifests from a GLB container's
+        // JSON chunk and resource data from the binary chunk.
+        if (pathFileExt == MakePathExt(GLB_EXTENSION))
+        {
+            auto glbStream = streamReader->GetInputStream(pathFile.u8string()); // Pass a UTF-8 encoded filename to GetInputString
+            auto glbResourceReader = std::make_unique<GLBResourceReader>(std::move(streamReader), std::move(glbStream));
+
+            manifest = glbResourceReader->GetJson(); // Get the manifest from the JSON chunk
+
+            resourceReader = std::move(glbResourceReader);
+        }
+
+        if (!resourceReader)
+        {
+            throw std::runtime_error("Command line argument path filename extension must be .gltf or .glb");
+        }
+
+        Document document;
+
+        try
+        {
+            document = Deserialize(manifest);
+        }
+        catch (const GLTFException& ex)
+        {
+            std::stringstream ss;
+
+            ss << "Microsoft::glTF::Deserialize failed: ";
+            ss << ex.what();
+
+            throw std::runtime_error(ss.str());
+        }
+
+        std::cout << "### glTF Info - " << pathFile << " ###\n\n";
+
+        PrintDocumentInfo(document);
+        PrintResourceInfo(document, *resourceReader);
+    }
+}
+
+#if defined _WIN32 && defined _UNICODE
+int wmain(int argc, wchar_t* argv[])
+#else // _WIN32 & _UNICODE
+int main(int argc, char* argv[])
+#endif
+{
+    try
+    {
+        if (argc != 2U)
+        {
+            throw std::runtime_error("Unexpected number of command line arguments");
```

Comment:
```suggestion
            throw std::runtime_error("Unexpected number of command line arguments\n");
```

Without the newline the errors message looks strange when you run it on the command line.

Reply:
I've added a trailing newline in the catch block when the error message